src/page-modules/assistant/client/journey-planner/index.ts

```diff
@@ -0,0 +1,24 @@
+import { TripData } from '@atb/page-modules/assistant/server/journey-planner/validators';
+import { TripQueryObject } from '@atb/page-modules/assistant/';
+
+export type TripApiReturnType = TripData;
+
+export async function nextTripPatterns(
+  query: TripQueryObject,
+  cursor: string,
+): Promise<TripApiReturnType> {
+  const queryString = Object.keys(query)
+    .map(
+      (key) =>
+        encodeURIComponent(key) +
+        '=' +
+        encodeURIComponent(String(query[key as keyof TripQueryObject])),
+    )
+    .join('&');
```

Comment:
I think you can do this easier:

```suggestion
  const queryString = new URLSearchParams(query).toString()
```

Reply:
I would need to change the `TripQueryObject` to only contain strings to do so, but that makes sense. No need for duplicate types really. I'll update it 👍